lib/manager/npm/post-update/npm.ts

```diff
@@ -49,7 +49,7 @@ export async function generateLockFile(
     const preCommands = [installNpm, 'hash -d npm'];
     const commands = [];
     let cmdOptions = '';
-    if (postUpdateOptions?.includes('npmDedupe') || skipInstalls === false) {
+    if (artifactUpdateApproach === 'deep') {
       logger.debug('Performing node_modules install');
       cmdOptions += '--ignore-scripts --no-audit';
```

Comment:
Ha, so no option to allow scripts. That's maybe a issue for some of our discussions about postinstall scripts.

needs a condition like `if (!getAdminConfig().allowScripts || config.ignoreScripts)` from `pnpm`